bindings/src/main/java/io/pravega/storage/filesystem/FileSystemWrapper.java

```diff
@@ -49,14 +57,50 @@
             PosixFilePermission.GROUP_READ,
             PosixFilePermission.OTHERS_READ);
 
+    @Getter
+    private final Cache<String, FileChannel> readCache;
+    @Getter
+    private final Cache<String, FileChannel> writeCache;
+
+    /**
+     * Constructs new instance of FileSystemWrapper.
+     * @param readCacheSize Read cache size
+     * @param writeCacheSize Write cache size
+     */
+    public FileSystemWrapper(int readCacheSize, int writeCacheSize) {
+        readCache = CacheBuilder.newBuilder()
+                .maximumSize(FileSystemStorageConfig.READ_CACHE_SIZE.getDefaultValue())
+                .removalListener(this::removeChannel)
+                .build();
+        writeCache = CacheBuilder.newBuilder()
+                .maximumSize(FileSystemStorageConfig.WRITE_CACHE_SIZE.getDefaultValue())
+                .removalListener(this::removeChannel)
+                .build();
+    }
+
+    void removeChannel(RemovalNotification<String, FileChannel> notification) {
```

Comment:
what you try to do here is close the channel instead of removing it. the channel removal will happen at the time when you call 
```
readCache.invalidate(path.toString());
writeCache.invalidate(path.toString());
```
I think the `closeChannel` is better :)

Reply:
May be `removeEntry` or even `handleRemovalNotification` is probably better as it is neutral enough without implying any implementation details.

Reply:
The following requested changes were made. Kindly review those. Thank you.